src/Common/src/System/Text/DBCSDecoder.cs

```diff
@@ -103,7 +104,7 @@ public unsafe override int GetCharCount(byte* bytes, int count, bool flush)
             if (count < 0)
                 throw new ArgumentOutOfRangeException(nameof(count), SR.ArgumentOutOfRange_NeedNonNegNum);
 
-            bool excludeLastByte = count > 0 || (!flush && IsLastByteALeadByte(bytes, count));
+            bool excludeLastByte = count > 0 && !flush && IsLastByteALeadByte(bytes, count);
```

Comment:
will try to think how to add a test for such case. the issue is we need to set the codepage on the console before requesting any encoding functionality . so we'll need to write some test as the one written in System.Diagnostics.Process